packages/cli-utils/src/commands/turbo/thread.ts

```diff
@@ -103,13 +104,22 @@ function collectImportsFromSourceFile(
         const importClause = node.getFullText().trim();
         if (turboOutputPath) {
           // Adjust the import specifier to point to the turbo output path
-          const adjustedSpecifier = resolveModuleName(
+          let adjustedSpecifier = resolveModuleName(
             specifier,
             sourceFile.fileName,
             turboOutputPath
-          )
-            .replace(/\.ts$/, '')
-            .replace(/\.tsx$/, '');
+          );
+
+          if (shouldTreatImportsAsNodeNext) {
+            if (adjustedSpecifier.endsWith('.ts') || adjustedSpecifier.endsWith('.tsx')) {
+              adjustedSpecifier = adjustedSpecifier
+                .replace(/\.ts$/, '.js')
+                .replace(/\.tsx$/, '.js');
+            }
```

Comment:
I'm a bit concerned about this, but I can investigate a replacement separately by looking at what other implementors of TS on-the-fly transpilation do

Reply:
afaik, there's no guarantee that the file will be resolved as such due to cjs/mjs extensions. Mainly, we're resolving a type from a file path, so in theory TypeScript will have some information (or canonical API) to resolve the file in its loaded sources without us essentially having to match/guess the filepath. Hence, I'm a little concerned that it feels like we're trying to match or guess TS' required resolution behaviour here